home/tables/columns.py

```diff
@@ -257,3 +269,24 @@ def render(self, value: dict):
             "args": {"merge_subject": model_obj.name, "subject_id": model_obj.pk}
         }
         return format_html(column_html, **kwargs)
+
+class ProfileReviewsActionColumn(ActionColumn):
+    def render(self, value: dict):
+        request = value.pop("request")
+        model_obj = value.pop("model_obj")
+
+        ctx = {}
+        ctx.update(csrf(request))
+
+        review = {
+            "professor": escape(Professor.unfiltered.get(pk=model_obj.professor_id).name),
+            "rating": model_obj.rating,
+            "anonymous": model_obj.anonymous,
+            "course": None if not model_obj.course_id else {"id": model_obj.course_id, "name": Course.unfiltered.get(pk=model_obj.course_id).name},
+            "grade": None if not model_obj.grade else model_obj.grade,
+            "id": model_obj.pk,
+            "content": escape(model_obj.content)
+        }
```

Comment:
I haven't looked into this deeply, but this is sketch. I would expect `model_obj` to already be a `Review` here - why go through all this trouble instead of dumping `model_obj`?

[Django's serializers](https://docs.djangoproject.com/en/4.1/topics/serialization/) should have a built-in way to follow and dump foreign keys as well.

Reply:
By default, [django serializers](https://docs.djangoproject.com/en/4.1/topics/serialization/#natural-keys) don't actually forward foreign keys and m2m relationships; the pk the field is used instead and we are required to add "natural keys" to our model managers to make this work any other way. If you think this is worth the effort than what's currently written then I'll go ahead and do it.